### PR TITLE
Serialize service - Part 1

### PR DESCRIPTION
**Pull request by awalsh3**
This PR creates a BookSerializer and it's unit tests. This PR will contain no changes to the controller. The unit tests are the low level kind, that follow black box testing approach followed for any plain old Ruby object.

### CODE REVIEW

**Comment by awalsh3** [comment]
Now this PR uses this new class in the controller and adjusts the tests as needed.

File: app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -2,38 +2,27 @@ module Api
...
-          serialized_books << {
-            type: "books",
-            id: book.id.to_s,
-            attributes: {
-              author_name: book.author_name,
-              title: book.title,
-              description: book.description,
-              rating: book.rating,
-              word_count: book.word_count
+      private
```
Comment (awalsh3) [inline app/controllers/api/v1/books_controller.rb]: why private?

Comment (patrickquigley102) [reply app/controllers/api/v1/books_controller.rb]: https://github.com/awalsh3/books/issues/19

**Comment by patrickquigley102** [comment]
Can you remove the controller changes from this PR and I'll review? As detailed in the associated the ticket.

We'll chat more about why this is good to do in the future :slightly_smiling_face:

**Comment by awalsh3** [comment]
> Can you remove the controller changes from this PR and I'll review? As detailed in the associated the ticket.
> 
> We'll chat more about why this is good to do in the future 🙂

This was tricky. Definitely learned my lesson to use a different branch...way easier.

**Review by patrickquigley102 — APPROVED** [review]
Lovely work here Annie, a lot of new concepts at once.

I can see the journey in the commits and I applaud it.

No feedback on this class today, I want us to focus on Exceptions still. We'll revisit this class in the future to talk about refactoring.

**Pull request merged**